plugin/evm/atomic/sync/atomic_block_sync_summary.go

```diff
@@ -21,6 +22,19 @@ var (
 	_ message.SyncableParser = (*AtomicSyncSummaryParser)(nil)
 )
 
+// codecWithAtomicSync is the codec manager that contains the codec for AtomicBlockSyncSummary and
+// other message types that are used in the network protocol. This is to ensure that the codec
+// version is consistent across all message types and includes the codec for AtomicBlockSyncSummary.
+var codecWithAtomicSync codec.Manager
```

Comment:
this was a little confusing to me since it's not used in the production code path, could we move it to the test file?

Reply:
I got even more confused why and how this was not a problem and put up a UT. Seems we don't really care about if a type is registered or not in linearcodec unless we marshal/unmarshal with an interface.